### PR TITLE
Add Pydantic validation to Serve AutoscalingConfig class

### LINKED ISSUE #19569: Add pydantic validation for Serve AutoscalingConfig options
Pydantic can automatically enforce conditions like `0 <= min_replicas <= max_replicas`.  We should also use `NonNegativeFloat` or `PositiveFloat` for values that represent times.

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

These changes validate `AutoscalingConfig`'s options with `Pydantic` to ensure non-default options obey logical constraints (e.g. `max_replicas >= min_replicas`). Also, `AutoscalingConfig` option types are now more precise (e.g. `PositiveInt` instead of `int`).

## Related issue number

Closes #19569.

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
      -  The `test_autoscaling_config_validation` test has been added to `test_config.py` to check that invalid options raise appropriate exceptions.

### CODE REVIEW

**Review by architkulkarni — APPROVED** [review]
Looks great! Thanks Shreyas!

File: python/ray/serve/config.py [hunk 1]
```diff
@@ -20,30 +21,37 @@ class AutoscalingConfig(BaseModel):
...
-    downscale_delay_s: float = 600.0
+    downscale_delay_s: NonNegativeFloat = 600.0
     # How long to wait before scaling up replicas
-    upscale_delay_s: float = 30.0
+    upscale_delay_s: NonNegativeFloat = 30.0
+
+    @validator("max_replicas")
+    def max_replicas_greater_than_or_equal_to_min_replicas(cls, v, values):
+        if "min_replicas" in values and v < values["min_replicas"]:
+            raise ValueError("max_replicas must be greater than"
```
Comment (architkulkarni) [inline python/ray/serve/config.py:52]: Nit: Should we print max_replicas and min_replicas in this message?

Comment (shrekris-anyscale) [reply python/ray/serve/config.py]: Good idea– I added their values to the error message.

**Comment by architkulkarni** [comment]
I think this is ready to merge @simon-mo

File: python/ray/serve/config.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import pydantic
 from google.protobuf.json_format import MessageToDict
 from pydantic import BaseModel, NonNegativeFloat, PositiveInt, validator
+from pydantic.types import NonNegativeInt, PositiveFloat
```
Comment (simon-mo) [inline python/ray/serve/config.py:9]: the import is weird, can we just import NonNegativeInt, PositiveFloat from pydantic instead of pydantic.types

Comment (shrekris-anyscale) [reply python/ray/serve/config.py]: Good call, I revised the imports.

**Pull request merged**